components/api/src/main/java/com/hotels/styx/api/StreamingHttpMessage.java

```diff
@@ -93,4 +95,16 @@ default boolean chunked() {
         return HttpMessageSupport.chunked(headers());
     }
 
+    default void consume() {
+        body().drop().aggregate(1)
+                .thenApply(buffer -> {
+                    buffer.delegate().release();
+                    return null;
+                });
+    }
+
+    default void drop() {
```

Comment:
No because the `drop` on the message still consumes the byte stream arriving from the TCP connection. It just drops each byte chunk that is received while retaining the publisher chain between styx server and styx client.

This message on the other hand drops the message. It is saying that we want to drop this message as it is and we are not interested in consuming the content. 

As a result the underlying TCP connection gets closed.

Reply:
So this method must be used before any of the content is read?

Reply:
Let's remove this from the API for now, and make a note of this issue.

Basically the only way to "read" content before calling this method is to subscribe with a 3rd partly subscriber like so:

```
        flux = Flux.from(response.body())
                .subscribe(buf -> doSomething(buf))
```

So now it would be illegal to call `drop` while this subscription is ongoing. This is because it causes a 2nd subscription to the live data stream which is not supported.